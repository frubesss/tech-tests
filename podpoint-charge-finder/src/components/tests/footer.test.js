import React from 'react';
import Footer from '../footer'
import renderer from 'react-test-renderer'
import {shallow} from 'enzyme'

describe('Footer Component', () => {

    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Footer buttonClick={() => {}}/>);
    });

    it('Snapshot Test', () => {
        const tree = renderer.create(wrapper).toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe('Navigation Item', () => {

        it('should render with the favourites navigation item active', () => {
            expect(wrapper.find('.footer-nav').childAt(1).hasClass('active')).toBe(true);
        });

        describe('click', () => {

            beforeEach(() => {
                wrapper = shallow(<Footer buttonClick={() => {}} buttonActive={'MAP'}/>);
                wrapper.find('.footer-nav').childAt(0).simulate('click');
            });

            it('should change to the selected navigation item when a new navigation item is selected', () => {
                expect(wrapper.find('.footer-nav').childAt(0).hasClass('active')).toBe(true);
            });

            it('should remove active class from the previous navigation item when a new navigation item is selected', () => {
                expect(wrapper.find('.footer-nav').childAt(1).hasClass('active')).toBe(false);
            });

        });

    })

});
