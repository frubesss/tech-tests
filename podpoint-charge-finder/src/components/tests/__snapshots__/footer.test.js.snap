// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component Snapshot Test 1`] = `
<div
  className="footer-nav"
>
  <a
    className="footer-nav-item"
    onClick={[Function]}
  >
    <i
      className="fa fa-map-marker"
    />
    Map
  </a>
  <a
    className="footer-nav-item active"
    onClick={[Function]}
  >
    <i
      className="fa fa-star"
    />
    Favourites
  </a>
  <a
    className="footer-nav-item"
    onClick={[Function]}
  >
    <i
      className="fa fa-user-circle"
    />
    Account
  </a>
</div>
`;
